### PR TITLE
[WFCORE-5145] Elytron server-ssl-context allowed protocols

### PR DESCRIPTION
**Pull request by SoniaZaldana**
Jira: https://issues.redhat.com/browse/WFCORE-5145

To pass CI, this will require an Elytron upgrade. 

Analysis document: https://github.com/wildfly/wildfly-proposals/pull/338
Requires: https://issues.redhat.com/browse/ELY-2024
Elytron PR with tests: https://github.com/wildfly-security/wildfly-elytron/pull/1458
WildFly PR with tests and documentation updates: https://github.com/wildfly/wildfly/pull/13673

### CODE REVIEW

File: elytron/src/main/java/org/wildfly/extension/elytron/ElytronDescriptionConstants.java
Comment (fjuma) [inline elytron/src/main/java/org/wildfly/extension/elytron/ElytronDescriptionConstants.java:535]: Very minor nit, I think `SSL_V2_HELLO` reads a bit better than `SSL_V2HELLO`.

File: elytron/src/main/java/org/wildfly/extension/elytron/ElytronSubsystemTransformers.java
Comment (fjuma) [inline elytron/src/main/java/org/wildfly/extension/elytron/ElytronSubsystemTransformers.java:124]: Looks like the `ListRejectAttributeChecker` could be used here instead.

File: elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java
Comment (fjuma) [inline elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java:380]: It would also be good to add a test for the case where SSLv2Hello is only configured on the server side.

**Review by fjuma — CHANGES_REQUESTED** [review]

File: elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java
Comment (SoniaZaldana) [reply elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java:380]: Added

File: controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java
Comment (fjuma) [inline controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java:3634]: This can be removed now since it's unused.

File: elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java
Comment (fjuma) [reply elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java:380]: Looks like it would be good to add one more test case that makes use of the default protocols on the server side and the client side to ensure that SSLv2Hello isn't enabled by default.

Comment (fjuma) [reply elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java:380]: It would also be good to either tweak `testCommunication` or create a new method that could be used instead in order to assert that `serverSocket.getEnabledProtocols()`, `clientSocket.getEnabledProtocols()`, `serverSocket.getSession().getProtocol()`, and `clientSocket.getSession().getProtocol()` return the expected values.

**Review by fjuma — CHANGES_REQUESTED** [review]

File: elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java
Comment (SoniaZaldana) [reply elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java:398]: I've modified testCommunication to check the configured protocols.

File: elytron/src/test/resources/org/wildfly/extension/elytron/tls-ibm.xml
Comment (fjuma) [inline elytron/src/test/resources/org/wildfly/extension/elytron/tls-ibm.xml:114]: I think double check that the same additions made in `tls-sun.xml` are also made in `tls-ibm.xml` and `tls-oracle13plus.xml`.xml. Notice that in `TlsTestCase#prepare`, we can see that one of these three test files will be used depending on the JDK version. So for any resources that we're adding for new test cases, we need to make sure that we add the same ones to all three test files.

Comment (SoniaZaldana) [reply elytron/src/test/resources/org/wildfly/extension/elytron/tls-ibm.xml:114]: I've updated the files so that the same additions are added across all three files.

Comment (fjuma) [reply elytron/src/test/resources/org/wildfly/extension/elytron/tls-ibm.xml:114]: I think tls-ibm.xml is missing "SeverSslContextSSLv2HelloOpenSsl".

Comment (fjuma) [reply elytron/src/test/resources/org/wildfly/extension/elytron/tls-ibm.xml:114]: Actually, ignore that last comment, I can see that the test that makes use of that SSL context doesn't get run on IBM JDK. :)

**Review by fjuma — APPROVED** [review]
Looks good, thanks @SoniaZaldana!

File: elytron/src/main/java/org/wildfly/extension/elytron/ElytronSubsystemTransformers.java
Comment (fjuma) [inline elytron/src/main/java/org/wildfly/extension/elytron/ElytronSubsystemTransformers.java:137]: This block should be moved to the new `from13` method.

**Comment by jmesnil** [comment]
this PR is out of sync with the main branch.
I'm closing it. Please reopen it when it is up to date and can be built.

If it can not be built (when a component upgrade is missing), please open it as a draft.

**Comment by darranl** [comment]
Converted to draft whilst waiting for approval to include in the next Elytron tag.

**Comment by darranl** [comment]
@SoniaZaldana it looks like this one also has a couple of conflicts to address.

**Review by fjuma — APPROVED** [review]

**Pull request merged**
